### PR TITLE
Adding ODROID device support and some polished options

### PR DESCRIPTION
**Pull request by meveric**
Added some options to merge rapi2 branch into the armhf branch and make it more useable for other devices such as the ODROID boards.

Also added some options that might be useful for other devices as well, especially the system.keyboard and system.joystick option.

### CODE REVIEW

File: core/hw/mem/_vmem.cpp [hunk 1]
```diff
@@ -547,15 +547,30 @@ int ashmem_create_region(const char *name, size_t size)
...
+		//file in tmpfs would be better?
+		fd = -1;
+#else
 		fd = shm_open("/dcnzorz_mem", O_CREAT | O_EXCL | O_RDWR,S_IREAD | S_IWRITE);
 		shm_unlink("/dcnzorz_mem");
+#endif
 		if (fd==-1)
 		{
+#ifdef USE_VMEM_FILE
+			printf("Falling back to file allocation..\n");
```
Comment (kingosticks) [inline core/hw/mem/_vmem.cpp]: isn't this fine without the ifdef?

Comment (meveric) [inline core/hw/mem/_vmem.cpp]: true, not needed, just wanted to "revert" the changes that were done explicitly for the RPi2 so no one complains about it, but yes it's actually not needed.

File: shell/rapi2/Makefile [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+ifdef LTO_TEST
+	CXXFLAGS += -flto -fwhole-program 
+	LDFLAGS +=-flto -fwhole-program 
+endif
+
+INCS	:= -I/opt/vc/include/ -I/opt/vc/include/interface/vmcs_host/linux -I/opt/vc/include/interface/vcos/pthreads  -I$(RZDCY_SRC_DIR) -I$(RZDCY_SRC_DIR)/deps -I$(RZDCY_SRC_DIR)/khronos -I../linux-deps/include
+
+LIBS    :=  -L/opt/vc/lib/  -L../linux-deps/lib
+LIBS    += -lm -lrt -lEGL -lGLESv2 #-lglslcompiler -lIMGegl -lpvr2d -lsrv_um 
+LIBS    += -lpthread -lasound -lX11 -lXdmcp -lXau  -lbcm_host
```
Comment (kingosticks) [inline shell/rapi2/Makefile]: Don't need these x libraries

File: shell/rapi2/Makefile [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+LD=${CC}
+
+MFLAGS := -marm -march=armv7-a -mtune=cortex-a9 -mfpu=neon -mfloat-abi=hard -funroll-loops
+ASFLAGS := -march=armv7-a -mfpu=neon -mfloat-abi=softfp
+
+LDFLAGS	:=	-g -Wl,-Map,$(notdir $@).map,--gc-sections -Wl,-O3 -Wl,--sort-common 
+
+
+CXXFLAGS := -g -O3  -D RELEASE -c -D TARGET_BEAGLE -D TARGET_LINUX_ARMELv7
```
Comment (kingosticks) [inline shell/rapi2/Makefile]: Need to set `TARGET_RPI` too?

File: shell/rapi2/Makefile (hunk 2 above)
Comment (meveric) [inline shell/rapi2/Makefile]: Not sure, but that was done by @skmp he basicly just copied the Beagle Board Makefile and adapted it.
But yes I think it's not needed.

File: shell/rapi2/Makefile (hunk 3 above)
Comment (meveric) [inline shell/rapi2/Makefile]: I haven't touched the RPi Makefile, but yes, generally it would require -DUSE_VMEM_FILE and -DTARGET_RPI

**Comment by skmp** [comment]
Hmm, looks mostly good. I'd like to make the libbcm thing dynamic using dlopen -- this will unify the rapi path with the rest -- and gotta make sure that panda isn't affected.

Meanwhile, you might want to go over #455 - in short I need a permission from you to re-license. *me should work on a more up to date version of this and put it in the readme*

**Comment by skmp** [comment]
Will merge in as soon as I'm done w/ #655 and #642

**Pull request closed without merging**